src/Bridge/Doctrine/Orm/Util/QueryNameGenerator.php

```diff
@@ -11,175 +11,32 @@
-
-        while (null === $rootAlias) {
-            $mapping = $aliasMap[$targetAlias];
-
-            if ('root' === $mapping) {
-                $rootAlias = $targetAlias;
-            } else {
-                $associationStack[] = $mapping['association'];
-                $targetAlias = $mapping['parentAlias'];
-            }
-        }
-
-        $rootEntity = $rootEntities[array_search($rootAlias, $rootAliases)];
-
-        $rootMetadata = $managerRegistry
-            ->getManagerForClass($rootEntity)
-            ->getClassMetadata($rootEntity);
-
-        $metadata = $rootMetadata;
-
-        while (null !== ($association = array_pop($associationStack))) {
-            $associationClass = $metadata->getAssociationTargetClass($association);
-
-            $metadata = $managerRegistry
-                ->getManagerForClass($associationClass)
-                ->getClassMetadata($associationClass);
-        }
-
-        return $metadata;
-    }
-
-    /**
-     * Gets the relationship from a Join expression.
-     *
-     * @param Join $join
-     *
-     * @return string
-     */
-    public static function getJoinRelationship(Join $join) : string
-    {
-        static $relationshipProperty = null;
-        static $initialized = false;
-
-        if (!$initialized && !method_exists(Join::class, 'getJoin')) {
-            $relationshipProperty = new \ReflectionProperty(Join::class, '_join');
-            $relationshipProperty->setAccessible(true);
-
-            $initialized = true;
-        }
-
-        return (null === $relationshipProperty) ? $join->getJoin() : $relationshipProperty->getValue($join);
-    }
-
-    /**
-     * Gets the alias from a Join expression.
-     *
-     * @param Join $join
-     *
-     * @return string
-     */
-    public static function getJoinAlias(Join $join) : string
-    {
-        static $aliasProperty = null;
-        static $initialized = false;
-
-        if (!$initialized && !method_exists(Join::class, 'getAlias')) {
-            $aliasProperty = new \ReflectionProperty(Join::class, '_alias');
-            $aliasProperty->setAccessible(true);
-
-            $initialized = true;
-        }
-
-        return (null === $aliasProperty) ? $join->getAlias() : $aliasProperty->getValue($join);
-    }
-
-    /**
-     * Gets the parts from an OrderBy expression.
-     *
-     * @param OrderBy $orderBy
-     *
-     * @return string[]
-     */
-    public static function getOrderByParts(OrderBy $orderBy) : array
-    {
-        static $partsProperty = null;
-        static $initialized = false;
-
-        if (!$initialized && !method_exists(OrderBy::class, 'getParts')) {
-            $partsProperty = new \ReflectionProperty(OrderBy::class, '_parts');
-            $partsProperty->setAccessible(true);
-
-            $initialized = true;
-        }
-
-        return (null === $partsProperty) ? $orderBy->getParts() : $partsProperty->getValue($orderBy);
+        return sprintf('p_%s', $this->incrementedName++);
```

Comment:
What do you want something different ?

https://gist.github.com/Simperfit/b2dafc636d5cc81725efca7aa0b0a387

I does still make sense, if you read the query, the first part is clearly too long for nothing in return since we understand that only by reading the query.